### ISSUE TITLE
Support complex haystacks/documents

### ISSUE THREAD

**Issue by wyozi** [leeoniya/uFuzzy#41]
Hi, thanks for the great library! Been a pleasure to use :)

I have a usecase, where I have an array of objects (e.g. `Array<{ name: string; location: string }>`) and I would like to implement a search that finds values in every key and ranks them in some order. Preferably, each key would have a different weight used to determine search ranking.

Is there a simple way to achieve this currently or would it involve mashing up multiple search results together?

**Comment by wyozi** [comment]
And about two minutes later I found [this](https://github.com/leeoniya/uFuzzy/issues/7#issuecomment-1276777783) hah. It solves the problem nicely (at least without weights)

**Comment by leeoniya** [comment]
i dont think this lib would be a suitable replacement for an actual document search, and i have no plans to go therr. you can probably build something crude on top of uFuzzy like the example you linked, but at some point you need to build a real index to do things well/efficiently. FlexSearch and MiniSearch are pretty good, but a backend database like MeleiSearch or TypeSense is better ;)

this lib achieves exactly what it was built to do. any new features will be small/incremental and most future work here will be bugfixes.

cheers!
